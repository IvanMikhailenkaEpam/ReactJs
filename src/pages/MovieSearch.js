import React, {Component} from 'react';
import Navbar from '../components/navbar/Navbar';
import SortBarInfo from '../components/sort-bar/sort-bar-info/SortBarInfo';
import SearchResults from '../components/search-result/SearchResults';
import SearchArea from "../components/navbar/search-area/SearchArea";
import SortBar from "../components/sort-bar/SortBar";
import SortByOptions from "../components/sort-bar/sort-by-options/SortByOptions";
import {MovieSearchService} from "../serveses/move-search/MovieSearchService";
import {
  setSearchBy,
  setSearchValue,
  getFilmRequest,
  getFilmsRequest,
  searchFilmRequest,
  sortFilmByParam
} from "../store/movie-search/actions";
import connect from "react-redux/es/connect/connect";

const queryString = require("query-string");

class MovieSearch extends Component {

  componentDidMount() {
    const {searchValue, searchBy} = this.props;
    if (searchValue && searchValue.length > 0) {
      console.log("if");
      this.props.searchFilmRequest(searchValue, searchBy);
    } else {
      console.log("else");
      this.props.getFilmsRequest();
    }
  }

  handleSortByParamClicked = (event, sortParam) => {
    let {films} = this.props;
    this.props.sortFilmByParam(films, sortParam);
    this.forceUpdate()
  };


  render() {
    console.log("1")
    let {films, searchValue, searchBy} = this.props;
    let sortBarInfo = films.length + " films found";

    return (
      <React.Fragment>
        <Navbar>
          <SearchArea searchValue={searchValue} searchBy={searchBy}/>
        </Navbar>
        <SortBar>
          <SortBarInfo text={sortBarInfo}/>
          <SortByOptions onClick={this.handleSortByParamClicked}/>
        </SortBar>
        <SearchResults films={films}/>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  const query = queryString.parse(ownProps.location ? ownProps.location.search : "");
  return {
    searchValue: query.searchValue,
    searchBy: query.searchBy,
    films: state.movieSearch.films,
    film: state.movieSearch.film,
  }
};
const mapDispatchToProps = {
  setSearchValue: setSearchValue,
  getFilmsRequest: getFilmsRequest,
  searchFilmRequest: searchFilmRequest,
  getFilmRequest: getFilmRequest,
  sortFilmByParam: sortFilmByParam,
  setSearchBy: setSearchBy
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieSearch);
