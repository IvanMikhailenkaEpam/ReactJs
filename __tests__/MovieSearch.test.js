import React from 'react';
import { shallow } from 'enzyme';
import MovieSearch from "../src/pages/MovieSearch";

describe('MovieSearch', () => {
  it('renders MovieSearch correctly', () => {
    const component = shallow(<MovieSearch />);

    expect(component).toMatchSnapshot();
  });
});
