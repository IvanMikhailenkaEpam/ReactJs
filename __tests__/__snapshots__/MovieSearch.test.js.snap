// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearch renders MovieSearch correctly 1`] = `
<Fragment>
  <Navbar>
    <SearchArea />
  </Navbar>
  <SortBar>
    <SortBarInfo
      text="3 films found"
    />
    <SortByOptions
      onClick={[Function]}
    />
  </SortBar>
  <SearchResults
    films={
      Array [
        Object {
          "age": 1332,
          "description": "Some info",
          "director": "Tarantion 1",
          "duration": "122 min",
          "genre": "genre 1",
          "id": 1,
          "img": "https://m.media-amazon.com/images/M/MV5BNGNhMDIzZTUtNTBlZi00MTRlLWFjM2ItYzViMjE3YzI5MjljXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,686,1000_AL_.jpg",
          "name": "film 1",
          "rating": 4.5,
          "regards": "Oscar winning move",
        },
        Object {
          "age": 2332,
          "description": "Some info",
          "director": "Tarantion 2",
          "duration": "100 min",
          "genre": "genre 1",
          "id": 2,
          "img": "https://m.media-amazon.com/images/M/MV5BMDFkYTc0MGEtZmNhMC00ZDIzLWFmNTEtODM1ZmRlYWMwMWFmXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_.jpg",
          "name": "film 2",
          "rating": 4.6,
          "regards": "",
        },
        Object {
          "age": 3333,
          "description": "Some info",
          "director": "Tarantion 3",
          "duration": "175 min",
          "genre": "genre 3",
          "id": 3,
          "img": "https://m.media-amazon.com/images/M/MV5BNDE4OTMxMTctNmRhYy00NWE2LTg3YzItYTk3M2UwOTU5Njg4XkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SY1000_CR0,0,666,1000_AL_.jpg",
          "name": "film 3",
          "rating": 4.4,
          "regards": "Gold medal",
        },
      ]
    }
  />
</Fragment>
`;
